resources/idlharness.js

```diff
@@ -3394,16 +3407,66 @@ function idl_test(srcs, deps, idl_setup_func) {
     }, 'idl_test setup');
 }
 
-/**
- * fetch_spec is a shorthand for a Promise that fetches the spec's content.
- */
-function fetch_spec(spec) {
-    var url = '/interfaces/' + spec + '.idl';
+// TODO: it would be nice to support `idl_array.add_objects`
+function idl_test_shadowrealm(srcs, deps) {
+    const script_urls = [
+        "/resources/testharness.js",
+        "/resources/WebIDLParser.js",
+        "/resources/idlharness.js",
+    ];
+    promise_test(t => {
+        const realm = new ShadowRealm();
+        realm.evaluate("globalThis.self = globalThis; undefined");
+
+        return Promise.all(script_urls.map(url => fetch_text(url).then(s => realm.evaluate(s))))
+        .then(_ => Promise.all(srcs.concat(deps).map(fetch_spec)))
+        .then(specs => {
+            const idls = JSON.stringify(specs.map(i => i.idl));
+            const code = `
+                const idls = JSON.parse(${JSON.stringify(idls)});
```

Comment:
Why not just paste in `idls`? It will be a valid value in JS.

Reply:
So it is - left over from some older code, probably